src/hooks/useProductSummary.ts

```diff
@@ -0,0 +1,30 @@
+import { useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { toast } from 'react-toastify';
+import { fetchProduct } from '@/api/ProductApi';
+import type { ProductSummary } from '@/types/Product';
+
+export function useProductSummary(productId: string | undefined) {
+  const [product, setProduct] = useState<ProductSummary | null>(null);
+  const [loading, setLoading] = useState(true);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (!productId) return;
+
+    fetchProduct(productId)
+      .then(setProduct)
+      .catch((err) => {
+        const toastId = 'fetch-product-error';
+        if (!toast.isActive(toastId)) {
+          toast.error(err.message || '상품 정보를 불러올 수 없습니다.', {
+            toastId,
+          });
+        }
+        navigate('/');
+      })
+      .finally(() => setLoading(false));
+  }, [productId, navigate]);
+
+  return { product, loading };
+}
```

Comment:
여기서는 useApiRequest 를 사용하지 않았네요 ㅎㅎ
어떻게 사용하여 연결할 수 있을지 고민해보시면 좋겠어요.

Reply:
```suggestion
const toastId = 'fetch-product-error';

export function useProductSummary(productId: string | undefined) {
  const product = useApiRequest(fetchProduct, productId);
  const navigate = useNavigate();

  useEffect(() => {
    if (!product.error) {
      return;
    }
    toast.error(err.message || '상품 정보를 불러올 수 없습니다.', {
      toastId,
    });
    navigate('/');
  }, [data.error]);

  return data;
}
```

이렇게 사용할 수 있겠죠!?

useToastOnError 라는 훅을 만들어서 사용할 수도 있어요 ㅎㅎ

```suggestion
const toastId = 'fetch-product-error';

export function useProductSummary(productId: string | undefined) {
  const product = useApiRequest(fetchProduct, productId);
  const navigate = useNavigate();  
  useToastOnError({
    error: product.error,
    id: toastId,
    onError: () => navigate('/');
  });

  return data;
}
```